base/sparse/linalg.jl

```diff
@@ -290,6 +290,27 @@ A_ldiv_B!(U::UpperTriangular{T,<:SparseMatrixCSC{T}}, B::StridedVecOrMat) where
 (\)(L::LowerTriangular{T,<:SparseMatrixCSC{T}}, B::SparseMatrixCSC) where {T} = A_ldiv_B!(L, Array(B))
 (\)(U::UpperTriangular{T,<:SparseMatrixCSC{T}}, B::SparseMatrixCSC) where {T} = A_ldiv_B!(U, Array(B))
 
+function A_rdiv_B!(A::SparseMatrixCSC{T}, D::Diagonal{T}) where T
+    dd = D.diag
+    if (k = length(dd)) ≠ A.n
+        throw(DimensionMismatch("size(A, 2)=$(A.n) should be size(D, 1)=$k"))
+    end
+    nonz = nonzeros(A)
+    @inbounds for j in 1:k
+        ddj = dd[j]
+        if iszero(ddj)
+            throw(SingularException(j))
+        end
+        for k in nzrange(A, j)
```

Comment:
Right. We could dispatch to a slow naive setindex loop version if any of the elements are non finite.